cmd/krew/cmd/install.go

```diff
@@ -94,6 +102,21 @@ Remarks:
 				install = append(install, plugin)
 			}
 
+			if *manifestURL != "" {
+				resp, err := internal.DownloadFile(*manifestURL)
+				if err != nil {
+					return errors.Wrapf(err, "Error downloading manifest from %q", *manifestURL)
+				}
+				plugin, err := indexscanner.DecodePluginFile(resp)
+				if err != nil {
+					return errors.Wrap(err, "failed to load custom manifest file")
+				}
+				if err := validation.ValidatePlugin(plugin.Name, plugin); err != nil {
+					return errors.Wrap(err, "plugin manifest validation error")
+				}
+				install = append(install, plugin)
```

Comment:
this is too repetitive with the next block, it's not good for code cleanliness.

Reply:
True. 

he only alternative is to define `plugin` and `err` as global variables and then pass them to the repetitive block of code. I thought it would be a bad idea to declare variables with global scope and I preferred the repetitive method.

I can implement the alternative method as explained above if that makes sense.

Reply:
hmm, why don't we make `decodepluginfile` work on io.ReadCloser?

Both resp.Body and os.OpenFile return io.ReadCloser.

Reply:
```
var plugin index.Plugin
var gerr error

if *manifestURL != "" {
	resp, err := internal.DownloadFile(*manifestURL)
	if err != nil {
		return errors.Wrapf(err, "Error downloading manifest from %q", *manifestURL)
	}
	plugin, gerr = indexscanner.DecodePluginFile(resp)
} else if *manifest != "" {
	plugin, gerr = indexscanner.ReadPluginFile(*manifest)
}

if gerr != nil {
	return errors.Wrap(gerr, "failed to load custom manifest file")
}
if err := validation.ValidatePlugin(plugin.Name, plugin); err != nil {
	return errors.Wrap(err, "plugin manifest validation error")
}
install = append(install, plugin)
```

How does this look? This snippet will replace this piece of code, https://github.com/kubernetes-sigs/krew/blob/6df65f26ed4b6420ad8ed0678ddb4cfe0a0ad99e/cmd/krew/cmd/install.go#L105-L129

Reply:
good enough but instead of creating `gerr`, you should probably wrap/return errors immediately where they occur.

Ideally I'd love to see us using `indexscanner.Decode` just accept `io.ReadCloser`. Maybe we can do it later, not in this PR.

Reply:
Done